src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -114,6 +114,42 @@ public PersistentDataStore() {
     return conversations;
   }
 
+	  /**
+   * Loads the activity feed's Conversation objects from the Datastore service and returns it
+   *
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *     Datastore service
+   */
+  public Conversation loadActFeedConversation() throws PersistentDataStoreException {
```

Comment:
Would you mind explaining me what this method does, I'm having a bit of trouble understanding?

Reply:
This retrieves the unique "activity feed" conversation from the place that it's stored in datastore. Whereas you usually retrieve the normal "list" (I don't think datastore actually returns an actual java list, but something like it that can be iterated through) of conversations using the key "chat-conversations" here you retrieve the activity feed conversation with the "act-conversation" key word. What this actually returns is some sort of table that one has iterate through in order to get the parameters of each "Entity" (Conversation/Message/User) that is stored there (there should only be one conversation stored there in this case). Once those parameters are retrieved, you make a new conversation object with those parameters (or if nothing is retrieved you create a Conversation with everything set to null since you're forced to instantiate the conversation you return). That new Conversation is eventually returned to wherever it was called from.

Reply:
I basically did this so that the activity feed conversation would be stored apart from the rest of the conversations, and wouldn't be pulled in when working with the rest of the conversations (like in the chatservlet, where I don't want the activity feed conversation popping up). Eventually it'd probably be nice to make an ActivityFeed object instead of a Conversation object and store that in datastore.

Reply:
I think it's loading the activityFeedConversation from the DataStore every time like how loadMessages() and loadUsers() is loading the messages and users?

Hm, I was wondering if we need to store the variables: uuid, ownerUuid, title, creationTime somewhere? since the for loop will generate all these values but we don't do anything with them yet :O Like in loadMessages() we add another instance to messages.add(m).. but not sure if we would need to do something like that for the case of activityFeedConversation?

Reply:
oh! I see you created the instance before returning at the end, so there should only be one activityFeedConversation, so maybe the for loop is unnecessary and we just need the try-catch?

Reply:
Yes I was thinking of trying to get rid of that for loop too, but what datastore.prepre(query) returns is some type of table/list. In this case there should only be one conversation ever stored within that table/list, but since the return type of datastore.prepare(query) is automatically a table/list of Entities I think you have to iterate through it in order to get what you want out, even though you just want the one thing.

Reply:
Ah I get what you mean, results.getIterable() is an iterable that needs to be iterated through, so even if we only have one item we still have to use the for loop. That looks good to me then!